firehydrant/errors.go

```diff
@@ -17,6 +18,46 @@ type APIError struct {
 	Messages []string `json:"messages"`
 }
 
+// UnmarshalJSON custom unmarshaler to handle messages field that can be either a string or array of strings
```

Comment:
Seems like we return `Messages` as a single string sometimes, so adding a custom unmarshall to handle